index.js

```diff
@@ -4,19 +4,26 @@
 
 'use strict';
 
-import { NativeModules, DeviceEventEmitter, NetInfo, Platform } from 'react-native';
+import { NativeModules, NativeEventEmitter, NetInfo, Platform } from 'react-native';
```

Comment:
Make Sure you're including DeviceEventEmitter, otherwise it throws an error. You're using it to add listeners, did you mean to add the listeners using NativeEventEmitter or DeviceEventEmitter?.